### ISSUE TITLE
"request" attribute of a connection when using asynchronous strategy

### ISSUE THREAD

**Issue by mikmatko** [cannatag/ldap3#288]
Thank you for this awesome library :)

When using a synchronous connection I can do something like this:

```python
with self.connection as conn:
   conn.search(base_dn, filter, attributes=attrs)
   conn.request
{'typesOnly': False, 'scope': 2, 'dereferenceAlias': 3, 'sizeLimit': 0, 'attributes': ['*'], 'controls': None, 'type': 'searchRequest', 'base': 'ou=People,dc=something,dc=something,dc=something', 'timeLimit': 0, 'filter': '(something=something)'}
```
The `request` attribute is something I like to have and store later for debugging purposes. However, as far as I can tell, you cannot access the attribute when using an asynchronous strategy. The `get_response()` method only returns the result and the response.

Would it be possible to have this functionality also when using an asynchronous strategy?

**Comment by cannatag** [comment]
This can be done. The request could be attached to the tuple returned in the get_response() with an optional parameter. 
May I ask why are you using the asynchronous strategy? Almost everybody use the synchronous one.

Bye,
Giovanni

**Comment by mikmatko** [comment]
Thanks, it would really help. :)

> May I ask why are you using the asynchronous strategy?

Started updating a project to use asyncio and the REUSABLE strategy seems like a good choice, until LDAP3 directly supports asyncio.

**Comment by cannatag** [comment]
I've added a get_request parameter to the get_response() method. If it's set to True the method returns a tuple in the format (response, result, request). 

Please check the code in dev.

**Comment by mikmatko** [comment]
Thank you. Seems to be working for ASYNC strategy, but not for REUSABLE. When using REUSABLE an empty dict is just returned. It seems like `get_response()` is internally called (before I'm calling it) when using REUSABLE.

**Comment by cannatag** [comment]
I know. I've worked on the Async strategy only. If you think this is the right approach I will work on the REUSABLE strategy too.

**Comment by mikmatko** [comment]
Can't say that there's anything wrong with it, I think it's the right approach :) Appreciate you taking the time to fix this. Thanks again!

**Comment by cannatag** [comment]
Check code in dev. get_request is honored in Reusable strategy too.

**Comment by mikmatko** [comment]
Seems to be working. Thank you! :+1:

**Comment by cannatag** [comment]
will be released in the next version of ldap3.

**Comment by cannatag** [comment]
released in v2.2.1
